plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -140,11 +140,14 @@ private static void sendPostRequest(String parametersString) {
       connection.setDoOutput(true);
       connection.setRequestMethod("POST");
       connection.setRequestProperty("Content-Length", Integer.toString(parametersString.length()));
+      connection.setReadTimeout(3000);  // milliseconds
 
       try (Writer writer = new OutputStreamWriter(connection.getOutputStream(), "UTF-8")) {
         writer.write(parametersString);
         writer.flush();
       }
+
+      connection.getResponseCode();  // This call is important to ensure posting the request.
```

Comment:
Yeah, I tried closing it, but it doesn't work. Looks like we need getResponseCode().